go/libkb/loaduser.go

```diff
@@ -360,3 +362,64 @@ func LoadUserPlusKeys(g *GlobalContext, uid keybase1.UID, cacheOK bool) (keybase
 
 	return up, nil
 }
+
+func checkForNewTracksIfSelf(arg LoadUserArg, idtable *IdentityTable) error {
+	arg.G().Log.Debug("+ checkForNewTracksIfSelf")
+	defer arg.G().Log.Debug("- checkForNewTracksIfSelf")
+
+	// If we're loading a user other than ourselves, short-circuit.
+	if !arg.Self || idtable == nil {
+		return nil
+	}
+
+	// Get the our last seen seqno from the local DB.
+	dbkey := DbKeyUID(DBLastSeenSeqno, arg.UID)
+	jw, err := arg.G().LocalDb.Get(dbkey)
+	var lastSeenSeqno Seqno
+	if jw == nil {
+		arg.G().Log.Debug("| no last seen seqno stored")
+	} else {
+		lastSeenSeqnoInt, err := jw.GetInt()
+		if err != nil {
+			return err
+		}
+		lastSeenSeqno = Seqno(lastSeenSeqnoInt)
+		arg.G().Log.Debug("| last seen seqno is %d", lastSeenSeqno)
+	}
+
+	// Find all the new track and untrack links since the last link we saw, and
+	// make a set of the users they refer to.
+	newSeenSeqno := lastSeenSeqno
+	changedUsers := make(map[keybase1.UID]bool)
+	for _, link := range idtable.links {
+		if link.GetSeqno() <= lastSeenSeqno {
+			continue
+		}
+		if link.GetSeqno() > newSeenSeqno {
+			newSeenSeqno = link.GetSeqno()
+		}
+		if trackChainLink, ok := link.(*TrackChainLink); ok {
```

Comment:
Can you use some sort of interface here, rather than type-casting for particular links?